src/main/java/com/icthh/xm/ms/entity/validator/EventDataTypeKeyValidator.java

```diff
@@ -0,0 +1,44 @@
+package com.icthh.xm.ms.entity.validator;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.icthh.xm.commons.exceptions.EntityNotFoundException;
+import com.icthh.xm.ms.entity.domain.Event;
+import com.icthh.xm.ms.entity.domain.XmEntity;
+import com.icthh.xm.ms.entity.domain.spec.EventSpec;
+import com.icthh.xm.ms.entity.service.XmEntitySpecService;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class EventDataTypeKeyValidator implements ConstraintValidator<EventDataTypeKey, Event> {
+
+    private final XmEntitySpecService xmEntitySpecService;
+
+    @Override
+    public boolean isValid(Event event, ConstraintValidatorContext constraintValidatorContext) {
+        XmEntity eventDataRef = event.getEventDataRef();
+        if (eventDataRef == null) {
+            return true;
+        }
+
+        String eventTypeKey = event.getTypeKey();
+        EventSpec eventSpec = xmEntitySpecService.findEvent(eventTypeKey)
+            .orElseThrow(() -> new EntityNotFoundException("Event specification not found by key: " + eventTypeKey));
+
+        String eventSpecDataTypeKey = eventSpec.getDataTypeKey();
+        if (isBlank(eventSpecDataTypeKey)) {
+            throw new IllegalStateException("Data type key not configured for Event with type key: "
+                + eventSpec.getKey());
+        }
+
+        if (xmEntitySpecService.getTypeSpecByKey(eventSpecDataTypeKey).isEmpty()) {
+            throw new EntityNotFoundException("Type specification not found by key: " + eventSpecDataTypeKey);
```

Comment:
i think better to return 400, not 404. @amedvedchuk , what your vision to this point?

Reply:
@abonert 
yes, agree with @sergeysenja1992 it is better to throw BusinessException (400). we will be able also to handle it in a common way (i.e. map code to translated message)

Reply:
@amedvedchuk @sergeysenja1992 Is it correct? `throw new BusinessException("xm.ms.entity.event.dataTypekey.notFound", "Type specification not found by key: " + eventSpecDataTypeKey);`

Reply:
@abonert for BusinessException you can use error code from the constants (com.icthh.xm.commons.exceptions.ErrorConstants)
if there is no such suitable constant you can define new.
(But i see there `error.validation` so i believe you can use it)

Reply:
During testing was determine that throwing exception inside validator is mistake, it leads to wrong handling via validation framework(JSR-380). Decided to enrich `ConstraintValidatorContext` with error information as it was already implemented in `com.icthh.xm.ms.entity.validator.JsonDataValidator`. Unfortunately current implementation of `com.icthh.xm.commons.i18n.error.web.ExceptionTranslator` always returns `error: error.validation, error_description: ${localized_message_by_error_code}` and there is no easy way to return error code that passed to exception

Reply:
pushed. Please [review](https://github.com/xm-online/xm-ms-entity/pull/217/commits/328657ea348c9c062515462d04883b0f9941ffb6)